relay/http.go

```diff
@@ -312,8 +336,17 @@ func newSimplePoster(location string, timeout time.Duration, skipTLSVerification
 	}
 }
 
-func (b *simplePoster) post(buf []byte, query string, auth string) (*responseData, error) {
-	req, err := http.NewRequest("POST", b.location, bytes.NewReader(buf))
+func (b *simplePoster) post(buf []byte, query string, auth string, q bool) (*responseData, error) {
+	loc := b.location
+	if q {
+		location, err := url.Parse(b.location)
```

Comment:
re-parsing a pre-computed value, then modifying the result to get a different string is definitely a poor design that would need to be removed